compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1354,6 +1354,8 @@ private boolean validateTableKeyValue(List<String> keySpecifierFieldNames,
                 dlog.error(recordLiteral.pos,
                         DiagnosticErrorCode.KEY_SPECIFIER_FIELD_VALUE_MUST_BE_CONSTANT_EXPR, fieldName);
                 data.resultType = symTable.semanticError;
+            }
+            if (data.resultType == symTable.semanticError) {
```

Comment:
It seems this is not a correct thing to do,
Consider this example.
```ballerina
table<FooRec> key(x, y) tb1 = table [
            {x: 1001, y: 20},
            {x: i, y: i}
        ];
```
(Note that there are two keys)
For this we are getting only one error, but we should get two errors for two keys.

I think the correct thing should be,
```java
    private boolean validateTableKeyValue(List<String> keySpecifierFieldNames,
                                                           List<BLangRecordLiteral> recordLiterals, AnalyzerData data) {

        for (String fieldName : keySpecifierFieldNames) {
            for (BLangRecordLiteral recordLiteral : recordLiterals) {
                BLangExpression recordKeyValueField = getRecordKeyValueField(recordLiteral, fieldName);
                if (recordKeyValueField != null && isConstExpression(recordKeyValueField)) {
                    continue;
                }

                dlog.error(recordLiteral.pos,
                        DiagnosticErrorCode.KEY_SPECIFIER_FIELD_VALUE_MUST_BE_CONSTANT_EXPR, fieldName);
                data.resultType = symTable.semanticError;
            }
        }

        return data.resultType != symTable.semanticError;
    }
```

With this change, if we have two key and spreadField, we are getting 4 errors(2 for one key). I think that is fine.
@lasinicl , @pcnfernando WDYT?